components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.ts

```diff
@@ -0,0 +1,110 @@
+import { set, pipe } from 'lodash/fp';
+import { LoadingStatus } from 'app/types/types';
+import {
+  ProjectsFilterActions,
+  ProjectsFilterActionTypes
+} from './projects-filter.actions';
+
+export interface ProjectsFilterOption {
+  label: string;
+  value: string;
+  checked: boolean;
+}
+
+export interface ProjectsFilterState {
+  options: ProjectsFilterOption[];
+  optionsLoadingStatus: LoadingStatus;
+  selectionLabel: string;
+  selectionCount: number;
+  selectionCountVisible: boolean;
+  selectionCountActive: boolean;
+}
+
+export const projectsFilterInitialState: ProjectsFilterState = {
+  options: [],
+  optionsLoadingStatus: LoadingStatus.notLoaded,
+  selectionLabel: 'All resources',
+  selectionCount: 0,
+  selectionCountVisible: false,
+  selectionCountActive: false
+};
+
+export function projectsFilterReducer(
+  state: ProjectsFilterState = projectsFilterInitialState,
+  action: ProjectsFilterActions): ProjectsFilterState {
+
+  switch (action.type) {
+
+    case ProjectsFilterActionTypes.LOAD_OPTIONS: {
+      return set('optionsLoadingStatus', LoadingStatus.loading, state);
+    }
+
+    case ProjectsFilterActionTypes.LOAD_OPTIONS_SUCCESS: {
+      return pipe(
+        set('options', action.payload),
+        set('optionsLoadingStatus', LoadingStatus.loadingSuccess),
+        set('selectionLabel', selectionLabel(action.payload)),
+        set('selectionCount', selectionCount(action.payload)),
+        set('selectionCountVisible', selectionCountVisible(action.payload)),
+        set('selectionCountActive', selectionCountActive(action.payload))
+      )(state) as ProjectsFilterState;
+    }
+
+    case ProjectsFilterActionTypes.LOAD_OPTIONS_FAILURE: {
+      return set('optionsLoadingStatus', LoadingStatus.loadingFailure, state);
+    }
+
+    case ProjectsFilterActionTypes.SAVE_OPTIONS: {
+      return pipe(
+        set('options', action.payload),
+        set('selectionLabel', selectionLabel(action.payload)),
+        set('selectionCount', selectionCount(action.payload)),
+        set('selectionCountVisible', selectionCountVisible(action.payload)),
+        set('selectionCountActive', selectionCountActive(action.payload))
+      )(state) as ProjectsFilterState;
+    }
+  }
+
+  return state;
+}
+
+function selectionLabel(options: ProjectsFilterOption[]): string {
+  const checked = options.filter(o => o.checked);
+
+  if (options.length === 1) {
+    return options[0].label;
+  }
+
+  if (checked.length === 1) {
+    return checked[0].label;
+  }
+
+  if (checked.length === options.length) {
+    return 'All projects';
+  }
+
+  if (checked.length > 1) {
+    return 'Multiple projects';
+  }
+
+  return 'All resources';
```

Comment:
Question for @susanev: I'm not sure the difference between `All Projects` and `All Resources` is clear. Maybe `Unfiltered` or `No Projects Selected`?

Reply:
So the difference between the two is only that _All Resources_ includes unassigned, since from the users perspective that is not a project. It does not have anything to do with whether a filter is applied or not. I haven't had a chance to review this yet though, so I will take a closer look later today.